### ISSUE TITLE
rt: Background threads might hang the program on panic

### ISSUE THREAD

**Issue by Finomnis** [tokio-rs/tokio#5433]
In the following code, a panic is created that propagates all the way out of `main`. But the tokio runtime prevents the program from exiting until the background thread is finished.

```rust
use std::{thread, time::Duration};
use tokio::{sync::oneshot, task};

fn main() {
    let runtime = tokio::runtime::Builder::new_current_thread()
        .enable_all()
        .build()
        .unwrap();

    runtime.block_on(async move {
        let (tx, rx) = oneshot::channel();
        task::spawn_blocking(move || {
            tx.send(()).unwrap();
            thread::sleep(Duration::from_secs(10));
        });
        rx.await.unwrap();

        panic!();
    });

    // This is never reached because `panic` will unwind
    // out of `block_on` and drop `runtime` directly
    runtime.shutdown_background();
}
```

## Expected
The program should exit immediately, because a clean shutdown is no longer possible anyway. In normal situations, waiting is desired (to make sure all `Drop`s have been executed peroperly).

## Potential solution
In the `Drop` implementation of the runtime or its background thread pool, it should be distinguished between normal drops and panics; in case of a panic, no waiting should be performed.

**Comment by Finomnis** [comment]
@Darksonn I would prefer this to be a config in the runtime builder instead of being completely rejected.

This is a real problem for me/my users: https://github.com/Finomnis/tokio-graceful-shutdown/issues/50

**Comment by Darksonn** [comment]
Your crate could define a wrapper around `Runtime` that invokes `shutdown_background()` in its destructor. Is that not sufficient for your use-case?

**Comment by Finomnis** [comment]
> Your crate could define a wrapper around `Runtime` that invokes `shutdown_background()` in its destructor. Is that not sufficient for your use-case?

@Darksonn Yeah I think that's what I did now. That is kind of misaligned with the goal of my crate to be as beginner friendly as possible though. But I think that's the outcome of our last conversation as well.

**Comment by Darksonn** [comment]
The problem is that if I change this behavior, then not waiting for background work can also be considered a bug. For example, many tests make use of temporary directories or similar that utilize destructors to clean up after themselves. Those cleanups statements may be skipped on panic with this change, which is not ideal.

**Comment by Finomnis** [comment]
@Darksonn Not if it's a flag in the runtime config that is disabled by default
